blackjack.el

```diff
@@ -676,46 +674,45 @@ Can be a single-character currency symbol such as \"$\", \"€\" or \"£\", or a
   "Return non-nil if DEALER-HAND up-card is an ace."
   (blackjack--is-ace-p (nth 1 (slot-value dealer-hand 'cards))))
 
+(defconst blackjack--card-value-hidden 13
+  "Token card \"value\" representing a hidden card.")
+
 (defun blackjack--draw-dealer-hand (game)
   "Draw the GAME dealer-hand."
-  (let* ((dealer-hand (slot-value game 'dealer-hand))
-         (cards (slot-value dealer-hand 'cards))
-         (hide-down-card (slot-value dealer-hand 'hide-down-card))
-         (card) (suit) (value))
+  (let ((dealer-hand (slot-value game 'dealer-hand)))
     (insert "  ")
-    (dotimes (x (length cards))
-      (setq card (nth x cards))
-      (if (and hide-down-card (= x 0))
-          (progn
-            (setq value 13)
-            (setq suit 0))
-        (setq value (slot-value card 'value))
-        (setq suit (slot-value card 'suit)))
-      (insert (blackjack--card-face game value suit))
-      (insert " "))
+    (seq-map-indexed
+     (lambda (card index)
+       "Display one of the cards in the dealer's hand."
+       (insert
+        (apply #'blackjack--card-face game
+               (if (and (= index 0) (slot-value dealer-hand 'hide-down-card))
+                   `(,blackjack--card-value-hidden 0)
```

Comment:
Right, because the suit does not matter on the dealer's down card.

Reply:
I may have got that right by accident then. I think I may have just forgotten to take care of setting the suit to zero.